client/my-sites/plans-v2/plans-column/index.tsx

```diff
@@ -35,7 +41,11 @@ interface PlanColumnType {
 	siteId: number | null;
 }
 
-type PlanWithBought = SelectorProduct & { owned: boolean; legacy: boolean };
+type PlanWithBought = SelectorProduct & {
+	owned: boolean;
+	legacy: boolean;
+	purchase: Purchase | null;
+};
```

Comment:
Instead of modifying this object, I'd recommend the code just pass `purchases` into the `PlanComponent` component and let it look things up from there.